janitor.go

```diff
@@ -0,0 +1,133 @@
+package fscache
+
+import (
+	"time"
+	"os"
+	"sort"
+	"fmt"
+)
+
+type janitorKV struct {
+	Key   string
+	Value fileStream
+}
+
+// Janitor is used to control when there are too many streams
+// or the size of the streams is too big.
+// It is called once right after loading, and then it is run
+// again after every Next() period of time.
+type Janitor interface {
```

Comment:
I think we can make Janitor a struct which implements Reaper.

To make this work, we'll need a little extra code, something like this maybe:

```go
interface StatsBasedReaper {
  ReapUsingStats(CacheStats ctx) (keysToReap []string)
}

struct CacheStats {
  // public fields / methods to provide info about the Cache state
  // this can expose the extra info needed by Janitor to decide what to reap
}
```
Some of the data you want to access through Janitor isn't available to the Cache through the FileSystem interface (like calling os.Stat won't work on an in memory FileSystem).

We could add another interface like:
```go
interface FileSystemStater {
  Stat(file stream.File) os.FileInfo
}
```
and then add the Stat method to the existing FileSystems to expose it if you use a type-assertion.

when Cache.haunt() get's called, if the Reaper implements "StatsBasedReaper" (checked with a type  assertion), then it can do something else instead of the Reaper loop, like:

```go
// inside c.haunt()
if cr, ok := c.grim.(StatsBasedReaper); ok {
  for _, key := range cr.ReapWithStats(CacheStats{cache: c}) {
    // reap the file with this key.
  }
  return
}

```